lib/std/fs.zig

```diff
@@ -301,7 +301,7 @@ pub const IterableDir = struct {
         .macos, .ios, .freebsd, .netbsd, .dragonfly, .openbsd, .solaris => struct {
             dir: Dir,
             seek: i64,
-            buf: [8192]u8, // TODO align(@alignOf(os.system.dirent)),
+            buf: [1024]u8, // TODO align(@alignOf(os.system.dirent)),
```

Comment:
I believe these bufs need to be at least PATH_MAX. Consider if a directory contains a file with the longest possible name. In such case, getdents would fail with EINVAL.

I suggest to put this code somewhere:

```zig
comptime assert(buf.len >= MAX_PATH_BYTES);
```

Reply:
I believe the relevant max length here is the max filename length (i.e. an individual component) rather than `PATH_MAX`. AFAICT the filename max length is 255 on Linux and 256 on Windows, so the reduced `buf` size should (hopefully) not cause as issue.

I'll investigate this more to be sure though, as I'm planning on working on https://github.com/ziglang/zig/issues/8268 as well which is directly related.

Reply:
Great point - in this case I think it's fine to merge this as is.

Reply:
does that mean lowering it to even 512 or 256 should be fine?

Reply:
@nektro 256 would be a problem since `@sizeOf(os.system.dirent)` is > 256 on Linux. 512 would work but would need to be benchmarked to ensure it doesn't regress performance (see https://github.com/ziglang/zig/pull/13073#discussion_r988415719).